pennylane_qiskit/remote.py

```diff
@@ -25,16 +25,95 @@ class RemoteDevice(QiskitDevice):
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from qiskit_ibm_runtime import QiskitRuntimeService
+
+        service = QiskitRuntimeService(channel="ibm_quantum")
+        backend = service.least_busy(n_qubits=127, simulator=False, operational=True)
+        dev = qml.device("qiskit.remote", wires=127, backend=backend)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=[0])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(1))
+
+    >>> circuit(np.pi/3, shots=1024)
+    0.529296875
+
+    This device also supports the use of local simulators such as ``AerSimulator`` or fake backends such as ``FakeManila``.
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from qiskit_aer import AerSimulator
+
+        backend = AerSimulator()
+        dev = qml.device("qiskit.remote", wires=5, backend=backend)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=[0])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(1))
+
+    >>> circuit(np.pi/3, shots=1024)
+    0.49755859375
+
+    We can also change the number of shots, either when initializing the device or when we execute
+    the circuit. Note that the shots number specified on circuit execution will override whatever
+    was set on device initialization.
+
+    .. code-block:: python
+
+        dev = qml.device("qiskit.remote", wires=5, backend=backend, shots=4000)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=[0])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(1))
+
+        circuit(np.pi/3) # this will run with 4000 shots
+        circuit(np.pi/3, shots=10000) # this will run with 10000 shots
+        circuit(np.pi/3) # this will run with 4000 shots
+
+    Internally, the device uses the `EstimatorV2 <https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.EstimatorV2/>`_
+    and the `SamplerV2 <https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.SamplerV2>`_  runtime primitives to execute
+    the measurements. To set options for `transpilation <https://docs.quantum.ibm.com/run/configure-runtime-compilation>`_
+    or `runtime <https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/options>`_, simply pass the keyword arguments into the device.
+    If you wish to change options other than ``shots``, PennyLane requires you to re-initialize the device to do so.
+
+    .. code-block:: python
+
+        import pennylane as qml
+        from qiskit_ibm_runtime.fake_provider import FakeManilaV2
+
+        backend = FakeManilaV2()
+        dev = qml.device("qiskit.remote", wires=5, backend=backend, resilience_level=1, optimization_level=1, seed_transpiler=42)
+        # to change options, re-initialize the device
+        dev = qml.device("qiskit.remote", wires=5, backend=backend, resilience_level=1, optimization_level=2, seed_transpiler=24)
```

Comment:
These lines are very long and don't as nicely in the docs (you get a long scroll bar in the code block). Could we add a line-break after each arg/kwarg being passed to `device` to keep the lines short?

Reply:
> It feels a little strange to update the docstring of the remote device to reflect changes to the new device, when it hasn't been switched over yet

I thought about this and I think either works for me. I wanted to rename the devices in this PR actually (if the changes weren't large) but it seems like there are some additional changes that warrant a separate PR. It does feel a little backwards to update the docstrings in this PR before renaming the devices so I think it's fine to revert this change here and just open a separate PR that will change all of it at the same time. Let me open that PR now so it's obvious.

Reply:
That sounds like extra work and I don't think it matters much, let's not worry about it :)